### ISSUE TITLE
On error, give access to body's data

### ISSUE THREAD

**Issue by MrDaves24** [kwhitley/itty-fetcher#39]
On the server side, I use the function error of itty-router. I give an error code and a custom body error. The body contains some hints about what's wrong. I use it to give a better UX on the frontend, accordingly to what happened in the backend.

Even tho I can see the object I sent in the body's response, I can't retrieve it when itty-fetcher throws an error without using a custom handleResponse.

Basically, my request is : When an error is thrown, give access to the body through the error.

On axios, it's done using err.response.data, but I don't want to use the huge axios where this is the only different thing I need

**Comment by kwhitley** [comment]
I agree - working on this right now!

**Comment by kwhitley** [comment]
I'm thinking of returning the content of the response assigned into the error itself.  If the content is text, this would be `err.message` instead.

So for instance the following error:
```js
400 {
  error: 'Missing values.',
  details: 'You are missing a valid username',
}
```

would throw an error with the following shape:
```js
{
  status: 400,                                   // automatic
  message: 'Bad Request',                        // automatic
  error: 'Missing values.',                      // from response.body
  details:  'You are missing a valid username',  // from response.body
  response,                                      // original response
}
```

**Comment by MrDaves24** [comment]
Is the full response object useful ?
I feel like it would use memory for not much

Maybe options like includeDetailsInError or includeResponseInError would be convenient

**Comment by kwhitley** [comment]
my only thought it it would give you access to the headers (which may actually be useful)

**Comment by kwhitley** [comment]
I'd also like to minimize options (preferring sane defaults) in order to minimize footprint of the library... this is already going to rob some precious bytes (but I agree... this would prevent a common step I do of passing a handleResponse option to intercept and rethrow a better error) :)

**Comment by kwhitley** [comment]
Discussion started at #41 :)

**Comment by kwhitley** [comment]
So for the sake of bytes alone, I think we'll leave out embedding the response itself back into the error.  Extending the StatusError to handle a private attribute (so we don't pollute deconstructions) added too many characters for the potential value.

If someone truly needs the original response, they can still take full control using the `handleResponse` option.

**Comment by kwhitley** [comment]
Live in `v0.9.x` 🚀
